components/ide/jetbrains/image/gha-update-image/lib/stable-version.test.ts

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+import { expect, test, mock, describe } from "bun:test";
+import { JetBrainsIDE, getStableVersionsInfo } from "./stable-version";
+
+describe("stableVersion", () => {
+    const testIdes: JetBrainsIDE[] = [
+        {
+            productName: "IntelliJ IDEA Ultimate",
+            productId: "intellij",
+            productCode: "IIU",
+            productType: "release",
+            exampleRepo: "https://github.com/gitpod-samples/spring-petclinic",
+        },
+        {
+            productName: "GoLand",
+            productId: "goland",
+            productCode: "GO",
+            productType: "release",
+            exampleRepo: "https://github.com/gitpod-samples/template-golang-cli",
+        },
+    ];
+
+    interface args {
+        IUBuildVersion: string;
+        IUMajorVersion: string;
+        GOBuildVersion: string;
+        GOMajorVersion: string;
+    }
+    interface expect {
+        err?: string;
+        buildVersion?: string;
+        majorVersion?: string;
+    }
+
+    const toAxiosData = (ide: JetBrainsIDE, majorVersion: string, buildVersion: string) => [
+        {
+            name: ide.productName,
+            link: ide.productCode,
+            releases: [
+                {
+                    majorVersion: majorVersion,
+                    build: buildVersion,
+                    downloads: {
+                        linux: { link: ide.productCode },
+                    },
+                },
+            ],
+        },
+    ];
+
+    test("getStableVersionsInfo", async () => {
+        const tests: { name: string; args: args; expect: expect }[] = [
+            {
+                name: "happy path",
+                args: {
+                    IUBuildVersion: "2024.1.1",
+                    IUMajorVersion: "2024.1",
+
+                    GOBuildVersion: "2024.1.1",
+                    GOMajorVersion: "2024.1",
+                },
+                expect: { err: undefined, buildVersion: "2024.1", majorVersion: "2024.1" },
+            },
+            {
+                name: "happy path with minimal build version",
+                args: {
+                    IUBuildVersion: "2024.1.300",
+                    IUMajorVersion: "2024.1",
+
+                    GOBuildVersion: "2024.1.200",
+                    GOMajorVersion: "2024.1",
+                },
+                expect: { err: undefined, buildVersion: "2024.1", majorVersion: "2024.1" },
```

Comment:
should buildversion be here 2024.1.200 ? not sure just asking

Reply:
buildVersion is processed inside tests so that I don't need to create a SemVer, will change to use SemVer